dev/consdeploy.py

```diff
@@ -0,0 +1,121 @@
+			return jsonRes
+        except urllib2.HTTPError as error:
+			print(error.read().decode())
+        except BaseException as e:
+			print(e)
+			return False
+    # Get repo's last tag
+    def getLastTag(self):
+        tags = self.post(self.getTagsUrl())
+        return tags[0]['name']
+    # Get repos last commit date
+    def getLastCommitDate(self):
+        commits = self.post(self.getCommitsUrl())
+        lastCommitDate = commits[0]['commit']['committer']['date']
+        return datetime.datetime.strptime(lastCommitDate, '%Y-%m-%dT%H:%M:%SZ')
+
+# Define constinuous deloyment class
+class ConsDeploy(object):
+    def __init__(self):
+        self.lastCommitDate = None
+        self.lastTag = None
+        self.gitaccess = GitHubAccess()
+        self.repoTag = None
+        self.repoCommitDate = None
+        self.interval = 2
+        self.repoBranch = "master"
+        self.devPath = "/opt/web/dev"
+        self.stagingPath = "/opt/web/staging"
+        self.devConfigPath = "_config_dev.yml"
+        self.stagingConfigPath = "_config_staging.yml"
+        return None
+    # Get local repo last commit date
+    def getLastCommitDate(self):
+        os.chdir(self.devPath)
+        os.chdir("source")
+        repoDateStr = subprocess.check_output("git show -s --format=%ci ", shell=True)
+        self.lastCommitDate = datetime.datetime.strptime(repoDateStr.split(" +")[0], '%Y-%m-%d %H:%M:%S')
+        print("Got last local commit date" + str(self.lastCommitDate))
+        return None
+    # Get local repo's latest tag
+    def getLastTag(self):
+        os.chdir(self.stagingPath)
+        os.chdir("source")
+        self.lastTag = subprocess.check_output("git describe --abbrev=0 --tags", shell=True)
+        print("Got last local tag: " + self.lastTag)
+        return None
+    # Get repo's last commit date
+    def getRepoCommitDate(self):
+        self.repoCommitDate = self.gitaccess.getLastCommitDate()
+        print("Got last repo commit date: " + str(self.repoCommitDate))
+        return None
+    # Get repo's latest tag
+    def getRepoTag(self):
+        self.repoTag = self.gitaccess.getLastTag()
+        print("Got last repo tag: " + self.repoTag)
+        return None
+    # Deploy for dev domain
+    def deployDev(self):
+        print("Start updating dev site!")
+        os.chdir(self.devPath)
+        os.chdir("source")
+        os.system("git pull origin " + self.repoBranch + " -f")
+        os.chdir(self.devPath)
+        os.system("jekyll build --config " + self.devConfigPath)
+        return None
+    # Deploy for staging domain
+    def deployStaging(self):
+        print("Start updating staging site!")
+        os.chdir(self.stagingPath)
+        os.chdir("source")
+        os.system("git pull origin " + self.repoBranch + " -f")
+        os.chdir(self.stagingPath)
+        os.system("jekyll build --config " + self.stagingConfigPath)
+        return None
+    # Start constinuous deployment
+    def consDeploy(self):
```

Comment:
Instead of keep this process running the whole time, use tool like cron makes more sense. And I am not sure if there is a way to restart it if this process accidentally crash.